.github/workflows/ci.yml

```diff
@@ -229,6 +231,27 @@ jobs:
       - name: "Run e2e tests"
         run: mvn ${{ env.MAVEN_CLI_OPTS }} verify -pl 'hxinsight-extension,e2e-test' -am -DskipUnitTests -DskipIntegrationTests -Dalfresco-platform.version=${{ matrix.repoVersion }} -Dalfresco-platform.java.version=${{ env.JAVA_VERSION_SUPPORTED_BY_REPO }}
 
+  nightly_tests:
+    name: "Run nightly tests"
+    runs-on: ubuntu-latest
+    if: github.event_name == 'schedule'
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v4
+      - name: "Set up JDK 17"
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: 'adopt'
```

Comment:
Why Adopt distribution? In Alfresco env Temurin is commonly used and it's set-up using this action: https://github.com/Alfresco/alfresco-build-tools/tree/master/.github/actions/setup-java-build

Reply:
Using above action is also safer in context of access to dependencies as it also set-up Maven `settings.xml`